src/drivers/imu/adis16477/ADIS16477.cpp

```diff
@@ -529,19 +454,38 @@ ADIS16477::write_reg16(uint8_t reg, uint16_t value)
 void
 ADIS16477::start()
 {
+#ifdef GPIO_SPI1_DRDY1_ADIS16477
+	// setup data ready on rising edge
+	stm32_gpiosetevent(GPIO_SPI1_DRDY1_ADIS16477, true, false, true, &ADIS16477::data_ready_interrupt, this);
```

Comment:
Please use the micro hal. https://github.com/PX4/Firmware/blob/master/src/platforms/px4_micro_hal.h#L216